src/AssetsController.ts

```diff
@@ -0,0 +1,265 @@
+	selectedAddress: string;
+}
+
+/**
+ * @type AssetsState
+ *
+ * Assets controller state
+ *
+ * @property allTokens - Object containing tokens per account and network
+ * @property allCollectibles - Object containing collectibles per account and network
+ * @property collectibles - List of collectibles associated with the active vault
+ * @property tokens - List of tokens associated with the active vault
+ */
+export interface AssetsState extends BaseState {
+	allTokens: { [key: string]: { [key: string]: Token[] } };
+	allCollectibles: { [key: string]: { [key: string]: Collectible[] } };
+	collectibles: Collectible[];
+	tokens: Token[];
+}
+
+/**
+ * Controller that stores assets and exposes convenience methods
+ */
+export class AssetsController extends BaseController<AssetsConfig, AssetsState> {
+	private getCollectibleApi(api: string, tokenId: number): string {
+		return `${api}${tokenId}`;
+	}
+
+	/**
+	 * Name of this controller used during composition
+	 */
+	name = 'AssetsController';
+
+	/**
+	 * List of required sibling controllers this controller needs to function
+	 */
+	requiredControllers = ['NetworkController', 'PreferencesController'];
+
+	/**
+	 * Creates a AssetsController instance
+	 *
+	 * @param config - Initial options used to configure this controller
+	 * @param state - Initial state to set on this controller
+	 */
+	constructor(config?: Partial<BaseConfig>, state?: Partial<AssetsState>) {
+		super(config, state);
+		this.defaultConfig = {
+			networkType: 'rinkeby',
+			selectedAddress: ''
+		};
+		this.defaultState = {
+			allCollectibles: {},
+			allTokens: {},
+			collectibles: [],
+			tokens: []
+		};
+		this.initialize();
+	}
+
+	/**
+	 * Adds a token to the stored token list
+	 *
+	 * @param address - Hex address of the token contract
+	 * @param symbol - Symbol of the token
+	 * @param decimals - Number of decimals the token uses
+	 * @returns - Current token list
+	 */
+	addToken(address: string, symbol: string, decimals: number) {
+		address = toChecksumAddress(address);
+		const { allTokens, tokens } = this.state;
+		const { networkType, selectedAddress } = this.config;
+		const newEntry: Token = { address, symbol, decimals };
+		const previousEntry = tokens.find((token) => token.address === address);
+		if (previousEntry) {
+			const previousIndex = tokens.indexOf(previousEntry);
+			tokens[previousIndex] = newEntry;
+		} else {
+			tokens.push(newEntry);
+		}
+		const addressTokens = allTokens[selectedAddress];
+		const newAddressTokens = { ...addressTokens, ...{ [networkType]: tokens } };
+		const newAllTokens = { ...allTokens, ...{ [selectedAddress]: newAddressTokens } };
+		const newTokens = [...tokens];
+		this.update({ allTokens: newAllTokens, tokens: newTokens });
+		return newTokens;
+	}
+
+	/**
+	 * Adds a collectible to the stored collectible list
+	 *
+	 * @param address - Hex address of the collectible contract
+	 * @param tokenId - The NFT identifier
+	 * @returns - Current collectible list
```

Comment:
Nit: This may be more accurate if it says "Promise resolving to the current collectible list".